h2/src/test/org/h2/test/unit/TestPgServer.java

```diff
@@ -695,4 +699,34 @@ private void testOtherPgClients() throws SQLException {
             conn0.close();
         }
     }
+
+    private void testArray() throws Exception {
+        if (!getPgJdbcDriver()) {
+            return;
+        }
+
+        Server server = createPgServer(
+                "-ifNotExists", "-pgPort", "5535", "-pgDaemon", "-key", "pgserver", "mem:pgserver");
+        try (
+                Connection conn = DriverManager.getConnection(
+                        "jdbc:postgresql://localhost:5535/pgserver", "sa", "sa");
+                Statement stat = conn.createStatement();
+        ) {
+            stat.execute("CREATE TABLE test (id int primary key, x1 int array, x2 varchar array)");
+            stat.execute("INSERT INTO test (id, x1, x2) VALUES (1, ARRAY[2, 3], ARRAY['4', '5'])");
```

Comment:
If you want to test `INTEGER ARRAY`, you need to add some support for it first. Currently all arrays are incorrectly mapped to `PG_TYPE_TEXTARRAY`, but they should be mapped to different types of PostgreSQL, this type is suitable only for `VARCHAR ARRAY`.

Reply:
Just test `VARCHAR ARRAY` only. Will add support for `INTEGER ARRAY` later. Some types may be added, like:

    public static final int PG_TYPE_BOOLARRAY = 1000; 
    public static final int PG_TYPE_INT2ARRAY = 1005; 
    public static final int PG_TYPE_INT4ARRAY = 1007;
    public static final int PG_TYPE_INT8ARRAY = 1016;
    public static final int PG_TYPE_FLOAT4ARRAY = 1021;
    public static final int PG_TYPE_FLOAT8ARRAY = 1022;
    ...